### PR TITLE
Fix regression causing signatures not to be rendered

### LINKED ISSUE #59: Regression with addMetaNote that effects time signatures, fermatas and tempo
I first noticed this with the bartok example where it's all in common time. Here's an example:

```Haskell
{-# LANGUAGE OverloadedStrings #-}

import Music.Prelude

main = defaultMain music

music = let 
    meta = title "Test 123"
        . composer "Some guy"
        . timeSignature (time 3 4)

    in meta $ pseq[c, d, e, f, g]
```

### PR DESCRIPTION
**Pull request by cianoc**
This fixes the regression. As far as I can tell fermata has never worked because there doesn't seem to be any code on the Lilypond side of things that generates a fermata.

In addition while looking at ```Meta.hs``` I found myself wondering if the ```Reactive``` should be replaced by a ```Behavior```. Currently fermatas are constructed using a duration of `0`, which on my understanding in a discrete set of events will be ignored. Using Behavior would solve that problem, though it's possible it would introduce new problems. An easy modification.

EDIT by @hanshoglund : This resolves the time signature aspect of #59

### CODE REVIEW

File: src/Music/Score/Meta/Time.hs [hunk 1]
```diff
@@ -203,8 +203,8 @@ isPowerOfTwo n = (n .&. (n -1)) == 0
 {-# INLINE isPowerOfTwo #-}
 
 -- TODO consolidate
-optionLast :: a -> Option (Last a)
-optionLast = Option . Just . Last
+optionLast :: a -> Maybe (Last a)
```
Comment (hanshoglund) [inline src/Music/Score/Meta/Time.hs]: Hi @cianoc 

Thank you very much! I have approved and feel free to merge. 

Quick question: #59 also mentions tempo, but this only seems to fix time signatures and fermata, is that correct?

**Review by hanshoglund — APPROVED** [review]

**Comment by hanshoglund** [comment]
> In addition while looking at Meta.hs I found myself wondering if the Reactive should be replaced by a Behavior. Currently fermatas are constructed using a duration of 0, which on my understanding in a discrete set of events will be ignored. Using Behavior would solve that problem, though it's possible it would introduce new problems.

It's actually the other way around: a Behaviors is continuous, so you can not observe instant events, but in a Reactive you can.

**Comment by cianoc** [comment]
Yes this doesn't fix either fermata or tempo. That seems to be caused by a different problem. From having looked at this again the problem seems to be that any Meta which takes an instantaneous time does not get attached to the score properly (so tempo and fermata). I thought that maybe that was related to the use of Reactive. I'm wondering if there's a regression in Reactive then that causes this. 

Though like I said, I could actually find any code that generates a fermata in Lilypond, so I'm not sure what's going on there.

**Comment by cianoc** [comment]
> > In addition while looking at Meta.hs I found myself wondering if the Reactive should be replaced by a Behavior. Currently fermatas are constructed using a duration of 0, which on my understanding in a discrete set of events will be ignored. Using Behavior would solve that problem, though it's possible it would introduce new problems.
> 
> It's actually the other way around: a Behavior is continuous, so you can not observe instant events, but in a Reactive you can 😃.

I'm not sure this is correct. If you have a discrete set of events they have to some length, even if it's tiny. If you had a sample rate (which reactive doesn't) you could kind of fudge using an event of length (1 sample). Whereas with a continuous signal you can have an impulse (which Behavior does), which only has a value at instant t, and has a value of 0 at all other times. Granted it's been a number of years since I did calculus, or DSP, but I think this is right :)

You could fudge an instant event for Reactive by making the duration incredibly small, but that's the kind of hack that could go wrong further down the line.

What seems to be happening with Reactive from looking at it is that an event of duration 0 is discarded - which seems like correct behavior to me. And behavior also _seems_ to do the right thing (though I'd need to do more investigation).

**Comment by hanshoglund** [comment]
@cianoc I added a regression test for the tempo issue in #69. It fails without this PR, as expected.

I don't think this actually has to do with Reactive vs. Behavior, but rather that `Score.Meta.Time.timeSignatureDuring` embeds the wrong type in the meta-data map. Will investigate further.

**Comment by hanshoglund** [comment]
I can confirm the cause:
- `Score.Meta.Time.timeSignatureDuring` embedding a meta-data item of type `Reactive (Option (Last TimeSignature))`
- `Score.Internal.Export.extractBarsInEra` now expects `Reactive (Last TimeSignature)`

This PR removes the now redundant `Option`  wrapper from `Score.Meta.Time`.

@cianoc Feel free to open another issue/PR if you think the documentation relating to meta-data or Reactive vs Behavior can be improved.

**Review by hanshoglund — APPROVED** [review]

**Comment by cianoc** [comment]
Oh thanks. I think I messed up the merge as I had fixed that.

**Comment by hanshoglund** [comment]
The fix merged here was correct, I just wanted to be sure about the cause. All good :).

**Pull request merged**